Ast/src/Lexer.cpp

```diff
@@ -332,24 +333,27 @@ void Lexer::setReadNames(bool read)
 
 const Lexeme& Lexer::next()
 {
-    return next(this->skipComments, true);
+    return next(this->skipTrivia, true);
 }
 
-const Lexeme& Lexer::next(bool skipComments, bool updatePrevLocation)
+const Lexeme& Lexer::next(bool skipTrivia, bool updatePrevLocation)
 {
-    // in skipComments mode we reject valid comments
+    // in skipTrivia mode we reject valid comments
     do
     {
-        // consume whitespace before the token
-        while (isSpace(peekch()))
-            consumeAny();
+        if (!FFlag::LuauLexerTokenizesWhitespace)
+        {
+            // consume whitespace before the token
+            while (isSpace(peekch()))
+                consumeAny();
+        }
 
         if (updatePrevLocation)
             prevLocation = lexeme.location;
 
         lexeme = readNext();
         updatePrevLocation = false;
-    } while (skipComments && (lexeme.type == Lexeme::Comment || lexeme.type == Lexeme::BlockComment));
+    } while (skipTrivia && (lexeme.type == Lexeme::Comment || lexeme.type == Lexeme::BlockComment || lexeme.type == Lexeme::Whitespace));
```

Comment:
Decided to merge `skipComments` together into a `skipTrivia` toggle

However, also open to splitting them into separate toggles if it makes more sense, e.g. `skipWhitespace`